src/lib.rs

```diff
@@ -22,41 +22,81 @@ use rand::Rng;
 use serde::{Deserialize, Serialize};
 use std::time::Duration;
 use testcontainers::{
-    clients,
+    clients::Cli,
     core::Port,
     images::generic::{GenericImage, Stream, WaitFor},
-    Container, Docker, Image,
+    Docker, Image,
 };
 
+pub use monerod::*;
+pub use wallet::*;
+
 /// How often we mine a block.
 const BLOCK_TIME_SECS: u64 = 1;
+
 /// Poll interval when checking if the wallet has synced with monerod.
 const WAIT_WALLET_SYNC_MILLIS: u64 = 1000;
 
-/// RPC client for `monerod` and `monero-wallet-rpc`.
+/// A docker container with a `monerod` instance and a `monero-wallet-rpc`
+/// instance.
 #[derive(Debug)]
-pub struct Client<'c> {
-    container: Container<'c, clients::Cli, GenericImage>,
-    pub wallet: wallet::Client,
-    pub monerod: monerod::Client,
+pub struct Container<'c> {
+    inner: testcontainers::Container<'c, Cli, GenericImage>,
+    pub monerod_rpc_port: u16,
+    pub wallet_rpc_port: u16,
 }
 
-impl<'c> Client<'c> {
-    /// Constructor that generates random port numbers for the local port
-    /// mapping of `monerod` and `monero-wallet-rpc`.
-    pub fn new(cli: &'c clients::Cli) -> Self {
+impl<'c> Container<'c> {
+    /// Creates a new monero container mapping random local ports for the two
+    /// RPC endpoints.
+    pub fn new(cli: &'c Cli) -> Self {
         let mut rng = rand::thread_rng();
         let monerod_rpc_port: u16 = rng.gen_range(1024, u16::MAX);
         let wallet_rpc_port: u16 = rng.gen_range(1024, u16::MAX);
 
-        let container = spin_up_container(cli, monerod_rpc_port, wallet_rpc_port);
+        let image = GenericImage::new("xmrto/monero")
```

Comment:
Instead of using `GenericImage`, you could also just implement the `Image` trait on a struct and pass that to `Docker::run` directly :)

That would obsolete this `Container` struct.

Once you've settled on an interface, you can upstream that to testcontainers then!

Reply:
Nice, will do this in a separate PR.